python_modules/dagster/setup.py

```diff
@@ -25,6 +26,45 @@ def get_version() -> str:
     return version["__version__"]
 
 
+def _get_grpcio_requires():
+    pypi_deps = [
+        # grpcio>=1.48.1 has hanging/crashing issues for python 3.9 and earlier:
+        # https://github.com/grpc/grpc/issues/30843
+        # ensure version we require is >= that with which we generated the grpc code (set in dev-requirements)
+        "grpcio>=1.32.0,<1.48.1; python_version < '3.10'",
+        "grpcio>=1.32.0; python_version >= '3.10'",
+    ]
+    if os.getenv("DAGSTER_M1_DEFAULT_GRPCIO"):
```

Comment:
can we fall back to this also if it's a py version that we don't have a wheel for? we should never end up with nothing installed

Reply:
added two lines for fallback on m1s (<3.8 and >=3.11)